storm-core/src/jvm/org/apache/storm/blobstore/LocalFsBlobStore.java

```diff
@@ -231,6 +231,18 @@ public void deleteBlob(String key, Subject who) throws AuthorizationException, K
     }
 
     @Override
+    public void clearRemovedBlob(String key) throws AuthorizationException {
+        validateKey(key);
+        // not pulling recent change and checking ACL
```

Comment:
@arunmahadevan 
Yes I was thinking about that way but over-thought about false-positive (when ZK is down, throwing Exception, etc). KeyNotFoundException clearly states that ZK is accessible and key is not exist so no false-positive. Thanks I'll address this.